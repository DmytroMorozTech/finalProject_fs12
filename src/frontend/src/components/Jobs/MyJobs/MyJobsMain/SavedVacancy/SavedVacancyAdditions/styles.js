import { makeStyles } from '@material-ui/core/styles'

export default makeStyles(() => ({

  menuItem: {
    '&:hover': {
      backgroundColor: 'rgba(0, 0, 0, 0.1)'
    }
  }
}))
