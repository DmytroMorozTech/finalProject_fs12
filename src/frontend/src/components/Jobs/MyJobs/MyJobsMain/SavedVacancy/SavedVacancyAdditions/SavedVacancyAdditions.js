import styles from './styles'
import React from 'react'
import TelegramIcon from '@material-ui/icons/Telegram'
import LinkIcon from '@material-ui/icons/Link'
import BookmarkBorderIcon from '@material-ui/icons/BookmarkBorder'
import SavedVacancyAdditionsItem from './SavedVacancyAdditionsItem/SavedVacancyAdditionsItem'
import { MenuItem } from '@material-ui/core'

function SavedVacancyAdditions () {
  const classes = styles()

  const handlerSendMessage = () => {
    console.log('Send in a message')
  }

  const handlerCopyLink = () => {
    console.log('Copy link')
  }

  const handlerUnsave = () => {
    console.log('Unsave')
  }

  const items = [
    {
      Icon: <TelegramIcon fontSize="inherit"/>,
      title: 'Send in a message',
      onClick: handlerSendMessage
    },
    {
      Icon: <LinkIcon fontSize="inherit"/>,
      title: 'Copy link',
      onClick: handlerCopyLink
    },
    {
      Icon: <BookmarkBorderIcon fontSize="inherit"/>,
      title: 'Unsave',
      onClick: handlerUnsave
    }
  ]

  return (
    <div>
      {items.map(({ Icon, title, onClick }, i) => (
        <MenuItem className={classes.menuItem}>
          <SavedVacancyAdditionsItem key={i} Icon={Icon} title={title} onClick={onClick}/>
        </MenuItem>
      ))}
    </div>
  )
}

export default SavedVacancyAdditions
