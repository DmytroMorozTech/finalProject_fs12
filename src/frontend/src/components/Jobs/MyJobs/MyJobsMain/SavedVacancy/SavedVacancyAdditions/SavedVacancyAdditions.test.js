import { shallow } from 'enzyme'
import React from 'react'
import SavedVacancyAdditions from './SavedVacancyAdditions'

const setUp = () => shallow(<SavedVacancyAdditions/>)

describe('should render SavedVacancyAdditions component', () => {
  let component
  beforeEach(() => {
    component = setUp()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
