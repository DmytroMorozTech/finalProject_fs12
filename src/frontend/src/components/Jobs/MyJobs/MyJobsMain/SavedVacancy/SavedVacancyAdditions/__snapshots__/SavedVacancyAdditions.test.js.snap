// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SavedVacancyAdditions component to match snapshot 1`] = `
<div>
  <WithStyles(ForwardRef(MenuItem))
    className="makeStyles-menuItem-1"
    key="0"
  >
    <SavedVacancyAdditionsItem
      Icon={
        <Memo
          fontSize="inherit"
        />
      }
      key="0"
      onClick={[Function]}
      title="Send in a message"
    />
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    className="makeStyles-menuItem-1"
    key="1"
  >
    <SavedVacancyAdditionsItem
      Icon={
        <Memo
          fontSize="inherit"
        />
      }
      key="1"
      onClick={[Function]}
      title="Copy link"
    />
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    className="makeStyles-menuItem-1"
    key="2"
  >
    <SavedVacancyAdditionsItem
      Icon={
        <Memo
          fontSize="inherit"
        />
      }
      key="2"
      onClick={[Function]}
      title="Unsave"
    />
  </WithStyles(ForwardRef(MenuItem))>
</div>
`;
